app/celery/tasks.py

```diff
@@ -952,9 +812,19 @@ def process_incomplete_job(job_id):
     template.process_type = db_template.process_type
 
     csv = get_recipient_csv(job, template)
-    for row in csv.get_rows():
+    rows = csv.get_rows()
+    first_row = []
+    for row in rows:
         if row.index > resume_from_row:
-            process_row(row, template, job, job.service)
+            first_row.append(row)
+            for result in chunked(rows, Config.BATCH_INSERTION_CHUNK_SIZE):
+                result = first_row + result
+                process_rows(result, template, job, job.service)
+                put_batch_saving_bulk_created(
+                    metrics_logger, 1, notification_type=db_template.template_type, priority=db_template.process_type
+                )
+                first_row = []
+            exit
```

Comment:
+1

Reply:
ok was going to meet Steve irl but I will try explaining this here. If it doesn't make sense let us meet irl.

l#817. = rows -> this is a generator
We need to loop through it till we get to the resume_from_row,
On l#820 we do that.

We then what to loop through the remaining part of "rows" in a chunk.
We want to exit the outside loop (it could be a break as well I think in this case).

Let us walk through an example
rows = [1, 2, 3, 4, 5, 6, 7, 8, 9, 10]
resume_from = 2
Config.BATCH_INSERTION_CHUNK_SIZE = 4
the first loop will yield from the generator until:
rows = [4, 5, 6, 7, 8, 9, 10]

but as we will have already yielded [3] and that should be included (as it is at row.index 2 and we want to resume from there) -> this is the first_row logic

Once that happens we want to then chunk `rows` using the batch_size of 4
This is why the for loop `exits` after `rows` has completely yielded everything. 

Let me know if we need to meet irl, happy to explain it.

Reply:
🎉 Thanks!